src/pytorch_lightning/utilities/xla_device.py

```diff
@@ -11,90 +11,36 @@
-    except Exception:
-        traceback.print_exc()
-        queue.put(None)
+    rank_zero_deprecation(
+        "`pytorch_lightning.utilities.xla_device.inner_f` has been deprecated in v1.8.0 and will be"
+        " removed in v1.10.0. Please use `lightning_lite.utilities.xla_device.inner_f` instead."
+    )
+    return new_inner_f(queue, func, *args, **kwargs)
 
 
 def pl_multi_process(func: Callable) -> Callable:
-    @functools.wraps(func)
-    def wrapper(*args: Any, **kwargs: Any) -> Union[bool, Any]:
-        queue: Queue = Queue()
-        proc = Process(target=inner_f, args=(queue, func, *args), kwargs=kwargs)
-        proc.start()
-        proc.join(TPU_CHECK_TIMEOUT)
-        try:
-            return queue.get_nowait()
-        except q.Empty:
-            traceback.print_exc()
-            return False
-
-    return wrapper
-
-
-class XLADeviceUtils:
-    """Used to detect the type of XLA device."""
-
-    _TPU_AVAILABLE = False
-
-    @staticmethod
-    @pl_multi_process
-    def _is_device_tpu() -> bool:
-        """Check if TPU devices are available.
-
-        Return:
-            A boolean value indicating if TPU devices are available
-        """
-        # For the TPU Pod training process, for example, if we have
-        # TPU v3-32 with 4 VMs, the world size would be 4 and as
-        # we would have to use `torch_xla.distributed.xla_dist` for
-        # multiple VMs and TPU_CONFIG won't be available, running
-        # `xm.get_xla_supported_devices("TPU")` won't be possible.
-        return (xm.xrt_world_size() > 1) or bool(xm.get_xla_supported_devices("TPU"))
-
-    @staticmethod
-    def xla_available() -> bool:
-        """Check if XLA library is installed.
-
-        Return:
-            A boolean value indicating if a XLA is installed
-        """
-        return _XLA_AVAILABLE
-
-    @staticmethod
-    def tpu_device_exists() -> bool:
-        """Runs XLA device check within a separate process.
-
-        Return:
-            A boolean value indicating if a TPU device exists on the system
-        """
-        if os.getenv("PL_TPU_AVAILABLE", "0") == "1":
-            XLADeviceUtils._TPU_AVAILABLE = True
-
-        if XLADeviceUtils.xla_available() and not XLADeviceUtils._TPU_AVAILABLE:
-
-            XLADeviceUtils._TPU_AVAILABLE = XLADeviceUtils._is_device_tpu()
-
-            if XLADeviceUtils._TPU_AVAILABLE:
-                os.environ["PL_TPU_AVAILABLE"] = "1"
-        return XLADeviceUtils._TPU_AVAILABLE
+    rank_zero_deprecation(
+        "`pytorch_lightning.utilities.xla_device.pl_multi_process` has been deprecated in v1.8.0 and will be"
+        " removed in v1.10.0. Please use `lightning_lite.utilities.xla_device.pl_multi_process` instead."
+    )
+    return new_pl_multi_process(func)
+
+
+class XLADeviceUtils(NewXLADeviceUtils):
+    def __init__(self) -> None:
+        rank_zero_deprecation(
+            "`pytorch_lightning.utilities.xla_device.XLADeviceUtils` has been deprecated in v1.8.0 and will be"
```

Comment:
This class isnt ever instantiated, so this deprecation wont help

I dont know why we create a class to define staticmethods when they could just be functions. The class seems to be an empty wrapper for the name

Reply:
Yes good point. For now, I added the deprecation to the two public static functions as well.